cmd/kar-controllers/app/metrics.go

```diff
@@ -0,0 +1,48 @@
+package app
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+    goroutines = promauto.NewGauge(prometheus.GaugeOpts{
+        Name: "mcad_goroutines_total",
+        Help: "Current number of goroutines",
+    })
+    memory = promauto.NewGauge(prometheus.GaugeOpts{
+        Name: "mcad_memory_usage_bytes",
+        Help: "Current memory usage",
+    })
+    gc = promauto.NewCounter(prometheus.CounterOpts{
+        Name: "mcad_gc_operations_total",
+        Help: "Total number of GC operations",
+    })
+    gcTime = promauto.NewCounter(prometheus.CounterOpts{
+        Name: "mcad_gc_time_seconds",
+        Help: "Total time spent in GC",
+    })
+)
+
+func RecordMetrics() {
```

Comment:
I'd strongly suggest we used _standard_ metrics provided by the Prometheus Process and Go collectors, rather than trying to reinvent the wheel.